src/frequenz/sdk/timeseries/battery_pool/_battery_pool_reference_store.py

```diff
@@ -0,0 +1,138 @@
+                battery.
+            power_manager_requests_sender: A Channel sender for sending power
+                requests to the power managing actor.
+            power_manager_bounds_subscription_sender: A Channel sender for sending
+                power bounds requests to the power managing actor.
+            min_update_interval: Some metrics in BatteryPool are send only when they
+                change. For these metrics min_update_interval is the minimum time
+                interval between the following messages.
+                Note that this argument is similar to the resampling period
+                argument in the ComponentMetricsResamplingActor. But as opposed to
+                ResamplingActor, timestamp returned in the resulting message will be
+                the timestamp of the last received component data.
+                It is currently impossible to use resampling actor for these metrics,
+                because we can't specify resampling function for them.
+            batteries_id: Subset of the batteries that should be included in the
+                battery pool. If None or empty, then all batteries from the microgrid
+                will be used.
+        """
+        if batteries_id:
+            self._batteries: frozenset[int] = frozenset(batteries_id)
+        else:
+            self._batteries = self._get_all_batteries()
+
+        self._working_batteries: set[int] = set()
+
+        self._update_battery_status_task: asyncio.Task[None] | None = None
+        if self._batteries:
+            self._update_battery_status_task = asyncio.create_task(
+                self._update_battery_status(batteries_status_receiver)
+            )
+
+        self._min_update_interval = min_update_interval
+
+        self._power_manager_requests_sender = power_manager_requests_sender
+        self._power_manager_bounds_subscription_sender = (
+            power_manager_bounds_subscription_sender
+        )
+
+        self._active_methods: dict[str, MetricAggregator[Any]] = {}
+        self._power_bounds_subs: dict[str, asyncio.Task[None]] = {}
+        self._namespace: str = f"battery-pool-{self._batteries}-{uuid.uuid4()}"
+        self._power_distributing_namespace: str = f"power-distributor-{self._namespace}"
+        self._channel_registry = channel_registry
+        self._formula_pool: FormulaEnginePool = FormulaEnginePool(
+            self._namespace,
+            self._channel_registry,
+            resampler_subscription_sender,
+        )
+
+    async def stop(self) -> None:
+        """Stop all pending async tasks."""
+        tasks_to_stop: list[Awaitable[Any]] = [
+            method.stop() for method in self._active_methods.values()
+        ]
+        tasks_to_stop.append(self._formula_pool.stop())
+        if self._update_battery_status_task:
+            tasks_to_stop.append(cancel_and_await(self._update_battery_status_task))
+        await asyncio.gather(*tasks_to_stop)
```

Comment:
And this.

I guess at this point it would delay the PR a bit, so I wouldn't block it because of this, but it would be good to have a follow-up PR converting this into a `BackgroundService`. The whole point of introducing that class was to unify how we build classes like this.

Reply:
made an issue: https://github.com/frequenz-floss/frequenz-sdk-python/issues/715